src/core/lv2/Lv2Proc.cpp

```diff
@@ -353,6 +363,14 @@ void Lv2Proc::copyBuffersToCore(sampleFrame* buf,
 void Lv2Proc::run(fpp_t frames)
 {
 	lilv_instance_run(m_instance, static_cast<uint32_t>(frames));
+
+	if (m_worker)
+	{
+		// Process any worker replies
+		m_worker->emitResponses();
```

Comment:
I can't see any restriction in the documentation as to when responses should be sent during processing. However, I imagine it would be more useful to the plugin if it could get the response before `lilv_instance_run`. (This probably has little effect since the response will be delayed somewhat anyway, but it should speed up delivery by one buffer duration in most cases.)

Reply:
I would also think that your suggestion is better, why fetch the worker's responses at the end of the cycle when we do not need them before next cycle? Your proposal gives the worker more time and should make it much more likely that the work is already finished.

Looking at ardour and jalv, they check the responses *after* "run". I asked @falkTX in IRC (libera, `#lv2`), and he said that the behavior between offline and online would be different if emitResponses comes before run. From how I understood him, if input is processed in "run" which leads to new "work", in offline mode, the worker replies would be fetched in the same cycle while in online mode, they would be fetched in the next cycle. While this is true, I cannot see how this would lead to different sound, since in both cases, the next audio will be generated in the next cycle. @falkTX please feel free to elaborate, we appreciate your opinion here.

Reply:
As far as I understand, there will always be a difference between offline and online mode. Offline responses are delivered immediately: the response comes before `lv2_instance_run`, and even `schedule_work`, return. Online responses are expected to be delivered later, which is the whole point of them - if the response could safely and reliably be delivered immediately, there would be no point submitting it to the worker. Accelerating the online response time usually by one buffer doesn't change this, and if anything makes the behaviour slightly closer to the instant response of offline mode.

I'm going to resolve this conversation for now, but I'm happy to continue it if someone has corrections or further input to provide.